src/serial/buf.cc

```diff
@@ -184,53 +178,55 @@ void Buf::EnsureRemainder(int length) {
   if ((forward_pos_ + length - 1) > reverse_pos_) {
     int new_size;
     if (length > 100) {
-      new_size = buf_.size() + length;
+      new_size = buf_->size() + length;
     } else {
-      new_size = buf_.size() + 100;
-    }
-    std::vector<uint8_t> new_buf(new_size);
-    for (int i = 0; i < forward_pos_; i++) {
-      new_buf.at(i) = buf_.at(i);
+      new_size = buf_->size() + 100;
     }
-    int reverse_size = buf_.size() - reverse_pos_ - 1;
-    int buf_start = reverse_pos_ + 1;
-    int new_buf_start = new_size - reverse_size;
+    int reverse_size = buf_->size() - reverse_pos_ - 1;
+    buf_->resize(new_size);
+    int new_reverse_tag = buf_->size() - 1;
+    int old_reverse_tag = new_reverse_tag - reverse_size;
     for (int i = 0; i < reverse_size; i++) {
-      new_buf.at(new_buf_start + i) = buf_.at(buf_start + i);
+      buf_->at(new_reverse_tag - i) = buf_->at(old_reverse_tag - i);
     }
     reverse_pos_ = new_size - reverse_size - 1;
-    buf_ = new_buf;
   }
 }
 
 std::string* Buf::GetBytes() {
```

Comment:
If no one use this function now, we could delete this function. Retun a pointer is not a good practice in production env.